src/controllers/capp.go

```diff
@@ -178,6 +189,71 @@ func (c *cappController) GetCappState(namespace, name string) (types.GetCappStat
 	return cappState, nil
 }
 
+func (c *cappController) GetCappDNS(namespace, name string) (types.GetDnsResponse, error) {
+	c.logger.Debug(fmt.Sprintf("Trying to fetch dns related to capp %q in namespace %q", name, namespace))
+
+	recordsK8S := &dnsrecordv1alpha1.CNAMERecordList{}
+
+	listOptions := prepareDNSListOptions(namespace, name)
+
+	err := c.client.List(c.ctx, recordsK8S, listOptions)
+	if err != nil {
+		c.logger.Error(fmt.Sprintf("Could not fetch dns related to capp %q in namespace %q with error: %v", name, namespace, err.Error()))
+		return types.GetDnsResponse{}, err
+	}
+
+	if len(recordsK8S.Items) == 0 {
+		_, err := c.GetCapp(namespace, name)
+		if err != nil {
+			return types.GetDnsResponse{}, err
+		}
+		return types.GetDnsResponse{}, nil
+	}
+
+	return types.GetDnsResponse{Records: parseDNSList(recordsK8S)}, nil
+}
+
+// parseDnsList converts dns from k8s object to the response dns type.
+func parseDNSList(recordsK8S *dnsrecordv1alpha1.CNAMERecordList) []types.DNS {
+	var records []types.DNS
+
+	for _, record := range recordsK8S.Items {
+		syncedStatus := corev1.ConditionFalse
+		readyStatus := corev1.ConditionFalse
+		dnsStatus := corev1.ConditionFalse
+
+		for _, condition := range record.Status.Conditions {
+			if condition.Type == xpv1.TypeReady {
+				readyStatus = condition.Status
+			} else if condition.Type == xpv1.TypeSynced {
+				syncedStatus = condition.Status
+			}
+		}
```

Comment:
You can do it in a slightly different way using the `GetCondition` method. See how it's done here:

https://github.com/dana-team/container-app-operator/blob/0d4e92c04f630d51ce941cf12523f8b473bf2258/internal/kinds/capp/utils/route.go#L33-L36

Reply:
when we talked you said to me that when condition not exists its equals for you False and not Unknown